json/src/maybe.rs

```diff
@@ -38,16 +38,19 @@ impl<T> MaybeEmptyVisitor<T> {
 impl<T> Visitor for MaybeEmptyVisitor<T> where T: Deserialize {
 	type Value = MaybeEmpty<T>;
 
-	fn visit_str<E>(&mut self, value: &str) -> Result<Self::Value, E> where E: Error {
+	fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+		write!(formatter, "an empty string or encoding of {}", stringify!(T))
```

Comment:
this just prints "an empty string or encoding of T" regardless of T. You're looking for `intrinsics::type_name` but it's unstable. https://github.com/rust-lang/rfcs/issues/1428